.github/actions/get-test-plan-key/action.yml

```diff
@@ -0,0 +1,75 @@
+name: get-test-plan-key
+description: "Determine the TP key if the TP exist, create it if doesn't"
+inputs:
+  major_version:
+    required: true
+    type: string
+  xray_token:
+    required: true
+    type: string
+outputs:
+  test_plan_key:
+    description: "the test plan key"
+    value: ${{ steps.get_test_plan_key.outputs.test_plan_key }}
+
+# NIGHTLY <OSS|MODULES|COLLECT> <MAJOR_VERSION> (e.g., NIGHTLY OSS 23.10)
+runs:
+  using: "composite"
+  steps:
+    - name: Get the TestPlanKey
+      id: get_test_plan_key
+      run: |
+        # Use a GraphQL query to get all test plans matching the JQL
+        graphql_query='{
+          "query":"query GetTestPlans($jql: String, $limit: Int!) { getTestPlans(jql: $jql, limit: $limit) { total results { issueId jira(fields: [\"summary\", \"key\"]) } } }",
+          "variables":{"jql": "project = MON","limit": 100}}'
+
+        # Execute the GraphQL query and parse the results
+        response=$(curl -H "Content-Type: application/json" -X POST -H "Authorization: Bearer ${{ inputs.xray_token }}" --data "$graphql_query" "https://xray.cloud.getxray.app/api/v2/graphql")
+
+        test_plans=$(echo "$response" | jq -c '.data.getTestPlans.results')
+
+        test_plan_key=$(node - <<EOF
+        const data = JSON.parse('$test_plans');
+        let test_plan_key = '';
+        data.forEach(item => {
+            const key = item.jira.key;
+            const summary = item.jira.summary;
+            if (summary === "NIGHTLY OSS ${{ inputs.major_version }}") {
+              test_plan_key = key;
+            }
+        });
+        console.log(test_plan_key);
+        EOF)
```

Comment:
Couldn't you use bash instead ?

Reply:
@sc979 unfortunately i couldn't :( , i tried to use jq command but the json result is not well formatted

Reply:
can you provide us an example of `.data.getTestPlans.results` please ?
to check with `jq` outside of the pipeline

Reply:
@kduret yes ofc :
https://github.com/centreon/centreon/actions/runs/6639197899/job/18037936008
https://github.com/centreon/centreon/actions/runs/6639595838/job/18039214707
https://github.com/centreon/centreon/actions/runs/6640036705/job/18040728876

Reply:
indeed it seems there are some special characters, but it works with : 
```
DATA='[{"issueId":"1","jira":{"key":"MON-1","summary":"Test plan 1"}},{"issueId":"2","jira":{"key":"MON-2","summary":"test plan 2"}}]'
echo $DATA | jq -c '.[].jira' | while read row; do
  echo "${row}" | jq '.key'
done
```